Docstring formatting

kevinsa5 opened the pull request:
<!-- Thank you for your contribution! The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items. Feel free to remove checklist items that are not relevant to your change. -->


 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Just a handful of changes to docstring text to improve sphinx rendering.  Do the changes for (non-new) functions need to be listed in whatsnew?

docstring | rendering issue | current link | new link
------------|--------|--------------|------------------
clearsky.ineichen | References section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.ineichen.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.ineichen.html)
clearsky.haurwitz | References section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.haurwitz.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.haurwitz.html)
ivtools.fit_sde_sandia | Returns section, Raises section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sde_sandia.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sde_sandia.html)
ivtools.fit_sdm_cec_sam | Returns section, Raises section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sdm_cec_sam.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sdm_cec_sam.html)
ivtools.fit_sdm_desoto | Returns section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sdm_desoto.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sdm_desoto.html)
tracking.singleaxis | Returns section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.tracking.singleaxis.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.tracking.singleaxis.html)
iotools.read_midc | Notes section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_midc.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_midc.html)
location.Location.from_tmy | Returns section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.location.Location.from_tmy.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.location.Location.from_tmy.html)
modelchain.ModelChain.run_model | Returns section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.modelchain.ModelChain.run_model.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.modelchain.ModelChain.run_model.html)
modelchain.ModelChain.complete_irradiance | Returns section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.modelchain.ModelChain.complete_irradiance.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.modelchain.ModelChain.complete_irradiance.html)
modelchain.ModelChain.prepare_inputs | Notes section | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.modelchain.ModelChain.prepare_inputs.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.modelchain.ModelChain.prepare_inputs.html)
api.rst | Double pvwatts_losses entry | [link](https://pvlib-python.readthedocs.io/en/latest/api.html#pvwatts-model) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/api.html#pvwatts-model)

Second round of edits for Reference/Footnote formatting:

docstring  | current link | new link
------------|----------------|------------------
solarposition.get_solarposition | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.get_solarposition.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.get_solarposition.html)
solarposition.spa_python | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.spa_python.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.spa_python.html)
solarposition.ephemeris | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.ephemeris.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.ephemeris.html)
solarposition.spa_c | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.spa_c.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.spa_c.html)
solarposition.sun_rise_set_transit_spa | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.sun_rise_set_transit_spa.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.sun_rise_set_transit_spa.html)
solarposition.sun_rise_set_transit_geometric | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.sun_rise_set_transit_geometric.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.sun_rise_set_transit_geometric.html)
solarposition.solar_zenith_analytical | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.solar_zenith_analytical.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.solar_zenith_analytical.html)
solarposition.solar_azimuth_analytical | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.solar_azimuth_analytical.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.solar_azimuth_analytical.html)
solarposition.declination_spencer71 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.declination_spencer71.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.declination_spencer71.html)
solarposition.declination_cooper69 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.declination_cooper69.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.declination_cooper69.html)
solarposition.equation_of_time_spencer71 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.equation_of_time_spencer71.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.equation_of_time_spencer71.html)
solarposition.equation_of_time_pvcdrom | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.equation_of_time_pvcdrom.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.equation_of_time_pvcdrom.html)
solarposition.hour_angle | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.hour_angle.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.hour_angle.html)
clearsky.ineichen | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.ineichen.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.ineichen.html)
clearsky.haurwitz | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.haurwitz.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.haurwitz.html)
clearsky.detect_clearsky | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.detect_clearsky.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.detect_clearsky.html)
clearsky.bird | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.bird.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.bird.html)
atmosphere.get_absolute_airmass | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.get_absolute_airmass.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.get_absolute_airmass.html)
atmosphere.get_relative_airmass | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.get_relative_airmass.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.get_relative_airmass.html)
atmosphere.pres2alt | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.pres2alt.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.pres2alt.html)
atmosphere.alt2pres | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.alt2pres.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.alt2pres.html)
atmosphere.bird_hulstrom80_aod_bb | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.bird_hulstrom80_aod_bb.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.bird_hulstrom80_aod_bb.html)
atmosphere.kasten96_lt | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.kasten96_lt.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.kasten96_lt.html)
atmosphere.angstrom_aod_at_lambda | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.angstrom_aod_at_lambda.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.angstrom_aod_at_lambda.html)
irradiance.get_extra_radiation | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.get_extra_radiation.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.get_extra_radiation.html)
irradiance.get_ground_diffuse | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.get_ground_diffuse.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.get_ground_diffuse.html)
irradiance.isotropic | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.isotropic.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.isotropic.html)
irradiance.perez | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.perez.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.perez.html)
irradiance.haydavies | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.haydavies.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.haydavies.html)
irradiance.klucher | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.klucher.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.klucher.html)
irradiance.reindl | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.reindl.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.reindl.html)
irradiance.dirint | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.dirint.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.dirint.html)
irradiance.dirindex | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.dirindex.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.dirindex.html)
irradiance.liujordan | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.liujordan.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.liujordan.html)
irradiance._get_perez_coefficients | None | None
iam.physical | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.physical.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.physical.html)
iam.ashrae | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.ashrae.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.ashrae.html)
iam.martin_ruiz | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.martin_ruiz.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.martin_ruiz.html)
iam.martin_ruiz_diffuse | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.martin_ruiz_diffuse.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.martin_ruiz_diffuse.html)
iam.sapm | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.sapm.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.sapm.html)
temperature.sapm_cell | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.temperature.sapm_cell.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.temperature.sapm_cell.html)
temperature.sapm_module | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.temperature.sapm_module.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.temperature.sapm_module.html)
temperature.pvsyst_cell | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.temperature.pvsyst_cell.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.temperature.pvsyst_cell.html)
pvsystem.calcparams_cec | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.calcparams_cec.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.calcparams_cec.html)
pvsystem.calcparams_desoto | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.calcparams_desoto.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.calcparams_desoto.html)
pvsystem.calcparams_pvsyst | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.calcparams_pvsyst.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.calcparams_pvsyst.html)
pvsystem.i_from_v | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.i_from_v.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.i_from_v.html)
pvsystem.singlediode | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.singlediode.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.singlediode.html)
pvsystem.v_from_i | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.v_from_i.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.v_from_i.html)
pvsystem.sapm | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.sapm.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.sapm.html)
pvsystem.sapm_effective_irradiance | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.sapm_effective_irradiance.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.sapm_effective_irradiance.html)
pvsystem.sapm_aoi_loss | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.sapm_aoi_loss.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.sapm_aoi_loss.html)
pvsystem.snlinverter | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.snlinverter.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.snlinverter.html)
singlediode.estimate_voc | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.singlediode.estimate_voc.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.singlediode.estimate_voc.html)
ivtools.fit_sde_sandia | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sde_sandia.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sde_sandia.html)
ivtools.fit_sdm_cec_sam | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sdm_cec_sam.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sdm_cec_sam.html)
ivtools.fit_sdm_desoto | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sdm_desoto.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sdm_desoto.html)
ivtools._system_of_equations_desoto | None | None
tracking.singleaxis | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.tracking.singleaxis.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.tracking.singleaxis.html)
iotools.read_tmy2 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_tmy2.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_tmy2.html)
iotools.read_tmy3 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_tmy3.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_tmy3.html)
iotools.read_epw | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_epw.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_epw.html)
iotools.read_srml | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_srml.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_srml.html)
iotools.read_srml_month_from_solardat | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_srml_month_from_solardat.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_srml_month_from_solardat.html)
iotools.read_surfrad | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_surfrad.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_surfrad.html)
iotools.get_psm3 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.get_psm3.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.get_psm3.html)
solarposition.get_solarposition | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.get_solarposition.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.get_solarposition.html)
solarposition.spa_python | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.spa_python.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.spa_python.html)
solarposition.ephemeris | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.ephemeris.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.ephemeris.html)
solarposition.spa_c | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.spa_c.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.spa_c.html)
solarposition.sun_rise_set_transit_spa | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.sun_rise_set_transit_spa.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.sun_rise_set_transit_spa.html)
solarposition.sun_rise_set_transit_geometric | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.sun_rise_set_transit_geometric.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.sun_rise_set_transit_geometric.html)
solarposition.solar_zenith_analytical | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.solar_zenith_analytical.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.solar_zenith_analytical.html)
solarposition.solar_azimuth_analytical | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.solar_azimuth_analytical.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.solar_azimuth_analytical.html)
solarposition.declination_spencer71 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.declination_spencer71.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.declination_spencer71.html)
solarposition.declination_cooper69 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.declination_cooper69.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.declination_cooper69.html)
solarposition.equation_of_time_spencer71 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.equation_of_time_spencer71.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.equation_of_time_spencer71.html)
solarposition.equation_of_time_pvcdrom | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.equation_of_time_pvcdrom.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.equation_of_time_pvcdrom.html)
solarposition.hour_angle | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.solarposition.hour_angle.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.solarposition.hour_angle.html)
clearsky.ineichen | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.ineichen.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.ineichen.html)
clearsky.haurwitz | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.haurwitz.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.haurwitz.html)
clearsky.detect_clearsky | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.detect_clearsky.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.detect_clearsky.html)
clearsky.bird | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.clearsky.bird.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.clearsky.bird.html)
atmosphere.get_absolute_airmass | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.get_absolute_airmass.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.get_absolute_airmass.html)
atmosphere.get_relative_airmass | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.get_relative_airmass.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.get_relative_airmass.html)
atmosphere.pres2alt | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.pres2alt.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.pres2alt.html)
atmosphere.alt2pres | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.alt2pres.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.alt2pres.html)
atmosphere.bird_hulstrom80_aod_bb | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.bird_hulstrom80_aod_bb.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.bird_hulstrom80_aod_bb.html)
atmosphere.kasten96_lt | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.kasten96_lt.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.kasten96_lt.html)
atmosphere.angstrom_aod_at_lambda | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.atmosphere.angstrom_aod_at_lambda.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.atmosphere.angstrom_aod_at_lambda.html)
irradiance.get_extra_radiation | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.get_extra_radiation.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.get_extra_radiation.html)
irradiance.get_ground_diffuse | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.get_ground_diffuse.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.get_ground_diffuse.html)
irradiance.isotropic | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.isotropic.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.isotropic.html)
irradiance.perez | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.perez.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.perez.html)
irradiance.haydavies | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.haydavies.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.haydavies.html)
irradiance.klucher | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.klucher.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.klucher.html)
irradiance.reindl | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.reindl.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.reindl.html)
irradiance.dirint | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.dirint.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.dirint.html)
irradiance.dirindex | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.dirindex.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.dirindex.html)
irradiance.liujordan | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.irradiance.liujordan.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.irradiance.liujordan.html)
irradiance._get_perez_coefficients | None | None
iam.physical | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.physical.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.physical.html)
iam.ashrae | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.ashrae.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.ashrae.html)
iam.martin_ruiz | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.martin_ruiz.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.martin_ruiz.html)
iam.martin_ruiz_diffuse | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.martin_ruiz_diffuse.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.martin_ruiz_diffuse.html)
iam.sapm | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iam.sapm.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iam.sapm.html)
temperature.sapm_cell | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.temperature.sapm_cell.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.temperature.sapm_cell.html)
temperature.sapm_module | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.temperature.sapm_module.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.temperature.sapm_module.html)
temperature.pvsyst_cell | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.temperature.pvsyst_cell.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.temperature.pvsyst_cell.html)
pvsystem.calcparams_cec | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.calcparams_cec.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.calcparams_cec.html)
pvsystem.calcparams_desoto | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.calcparams_desoto.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.calcparams_desoto.html)
pvsystem.calcparams_pvsyst | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.calcparams_pvsyst.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.calcparams_pvsyst.html)
pvsystem.i_from_v | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.i_from_v.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.i_from_v.html)
pvsystem.singlediode | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.singlediode.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.singlediode.html)
pvsystem.v_from_i | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.v_from_i.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.v_from_i.html)
pvsystem.sapm | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.sapm.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.sapm.html)
pvsystem.sapm_effective_irradiance | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.sapm_effective_irradiance.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.sapm_effective_irradiance.html)
pvsystem.sapm_aoi_loss | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.sapm_aoi_loss.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.sapm_aoi_loss.html)
pvsystem.snlinverter | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.snlinverter.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.pvsystem.snlinverter.html)
singlediode.estimate_voc | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.singlediode.estimate_voc.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.singlediode.estimate_voc.html)
ivtools.fit_sde_sandia | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sde_sandia.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sde_sandia.html)
ivtools.fit_sdm_cec_sam | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sdm_cec_sam.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sdm_cec_sam.html)
ivtools.fit_sdm_desoto | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.ivtools.fit_sdm_desoto.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.ivtools.fit_sdm_desoto.html)
ivtools._system_of_equations_desoto | None | None
tracking.singleaxis | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.tracking.singleaxis.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.tracking.singleaxis.html)
iotools.read_tmy2 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_tmy2.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_tmy2.html)
iotools.read_tmy3 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_tmy3.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_tmy3.html)
iotools.read_epw | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_epw.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_epw.html)
iotools.read_srml | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_srml.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_srml.html)
iotools.read_srml_month_from_solardat | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_srml_month_from_solardat.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_srml_month_from_solardat.html)
iotools.read_surfrad | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.read_surfrad.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.read_surfrad.html)
iotools.get_psm3 | [link](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.iotools.get_psm3.html) | [link](https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.iotools.get_psm3.html)

cwhanse commented:
> I am familiar with the contributing guidelines
> 
>   
>     
>   
>   
>     
>    Adds description and name entries in the appropriate "what's new" file in docs/sphinx/source/whatsnew for all changes. Includes link to the GitHub Issue with :issue:`num` or this Pull Request with :pull:`num`. Includes contributor name and/or GitHub username (link with :ghuser:`user`).
> 
>   
>     
>   
>   
>     
>    New code is fully documented. Includes numpydoc compliant docstrings, examples, and comments where necessary.
> 
>   
>     
>   
>   
>     
>    Pull request is nearly complete and ready for detailed review.
> 
>   
>     
>   
>   
>     
>    Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.
> 
> Just a handful of changes to docstring text to improve sphinx rendering.  Do the changes for (non-new) functions need to be listed in whatsnew?

Many thanks and especially for the well-formatted PR description! I can review.

Please add a note to the Documentation section of the whatsnew.

cwhanse commented on pvlib/modelchain.py:
```diff
@@ -924,6 +925,7 @@ def run_model(self, weather, times=None):
         -------
         self
 
+
```
Maybe move this up to the Extended summary position above the Parameters

kevinsa5 replied:
Changed -- I liked the way `prepare_inputs` lists these in the `Notes` section, so I've changed the other two method docstrings to do the same.

kevinsa5 commented on pvlib/iotools/midc.py:
```diff
@@ -180,9 +180,9 @@ def read_midc(filename, variable_map={}, raw_data=False, **kwargs):
-        e.g. If a MIDC file contains the variable 'Global Horizontal [W/m^2]',
-             passing the dictionary below will rename the column to 'ghi' in
-             the returned Dataframe.
+    e.g. If a MIDC file contains the variable 'Global Horizontal [W/m^2]',
```
Changed

kevinsa5 commented on pvlib/ivtools.py:
```diff
@@ -314,36 +312,34 @@ def fit_sdm_desoto(v_mp, i_mp, v_oc, i_sc, alpha_sc, beta_voc,
-            Optimization result of scipy.optimize.root().
-            See scipy.optimize.OptimizeResult for more details.
+    Dictionary with the following elements:
+        * `I_L_ref` : float
```
Changed.  I failed to find a way of getting sphinx to render these "sub-lists" nicely out of the box, so I've hardcoded in the formatting.  If someone knows a better way, I'd love to learn it.

cwhanse commented:
@wholmgren OK with me to merge

wholmgren commented on pvlib/clearsky.py:
```diff
@@ -65,22 +65,22 @@ def ineichen(apparent_zenith, airmass_absolute, linke_turbidity,
-        the Linke turbidity coefficient", Solar Energy, vol 73, pp. 151-157,
-        2002.
+    the Linke turbidity coefficient", Solar Energy, vol 73, pp. 151-157,
+    2002.
```
Neither the current format nor the proposed change is consistent with references section of [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)

kevinsa5 replied:
Fair, but I am having zero success with doing it the "right way" and getting the results to look any better than the ones here: https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.atmosphere.first_solar_spectral_correction.html

Is that how references are supposed to look?

wholmgren replied:
Hm... They *should* look like the format in the numpydoc example. I don't know why they do not. I poked at the html/css. I think numpydoc is writing out the correct rst and sphinx is processing it into the correct html, so I'm guessing it's related to the css. It could be specific to the readthedocs sphinx theme, but that's just a guess. 

So, none of the formats are ideal. I have a slight preference for sticking with the numpydoc format and hoping that someone, somewhere, fixes some bug for us. But I'm also ok with your "plain text" approach.

kevinsa5 replied:
> It could be specific to the readthedocs sphinx theme, but that's just a guess.

Good guess!  Looks like it's an incompatibility between sphinx_rtd_theme and sphinx 2.0+: https://github.com/readthedocs/sphinx_rtd_theme/issues/741.  It's a bit of a rabbit hole, but I've found two workarounds:

- Using sphinx <2.0 fixes the reference rendering, or at least v1.8.5 does for me locally.  Interestingly, RTD is installing 1.8.5 first and then replacing it with 2.2.2 -- see [the most recent RTD build log](https://readthedocs.org/api/v2/build/10102768.txt), ctrl-F "Successfully uninstalled Sphinx-1.8.5".  I'm guessing that the `sphinx=1.7.2` in the `environment.yml` file isn't being respected because that file is used for conda installs, but `readthedocs.yml` says to use pip and install all extra requirements, which according to `setup.py` includes sphinx without a version constraint.  So I'd guess that pinning sphinx at a pre-2.0 version in `setup.py` would change the version RTD ends up using.  RTD also encourages pinning specific versions of sphinx, see the blue `Note` section here: https://docs.readthedocs.io/en/stable/builds.html#understanding-what-s-going-on
- Apparently sphinx 2.0+ switched to HTML5 output, but the old HTML4 writer is [still available](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html4_writer).  I tried it out locally and it also fixes the references problem.

Both of these are a little heavy handed and have an effect on how the rest of the docs are rendered so I wanted to get feedback on whether this seemed like a good idea or if you'd prefer to just wait for the theme to fix it on their end.  Either way, I'll standardize the docstring references to the numpydoc format.  Some other functions already had the plaintext version (e.g. [clearsky.bird](https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.clearsky.bird.html)) so I'll change those as well.

cwhanse replied:
> standardize the docstring references to the numpydoc format.
+1 here.

wholmgren replied:
You are correct about the conda file being ignored. I’m ok with either work around.

wholmgren commented on pvlib/location.py:
```diff
@@ -99,8 +99,9 @@ def from_tmy(cls, tmy_metadata, tmy_data=None, **kwargs):
-        called this method from).
+        new_object
+            Location object (or the child class of Location that you
+            called this method from).
```
I'm guessing I wrote this a long time ago but I don't like it. Let's just make it

```
Returns
-------
Location
```

kevinsa5 commented:
I believe the remaining lint errors are from `tmy.py`.  I pinned sphinx at 1.8.5 and added a line to the sphinx conf to suppress warnings about unreferenced footnotes because the majority of docstrings with references don't actually cite them.  I added a new table in the original post but it's huge and it's easier to scroll through the changes here: https://github.com/pvlib/pvlib-python/pull/833/files

A couple other things I noticed: 
- This part of `api.rst` might have been overlooked when tmy.* was moved to iotools: https://pvlib-python.readthedocs.io/en/latest/api.html#tmy
- pvsystem.adrinverter doesn't have an entry in api.rst (other than through modelchain)

cwhanse commented:
@kevinsa5 randomly looking at some of the changes, "(1, 2, 3)" appears after "[1]" for the reference. Is this expected?  https://external-builds.readthedocs.io/html/pvlib-python/833/generated/pvlib.temperature.sapm_cell.html#id4

kanderso-nrel commented:
@cwhanse That one confused me too at first.  If the docstring text references a citation more than once, sphinx includes links to each reference location instead of using the `[1]` as the link.

cwhanse commented:
> @cwhanse That one confused me too at first.  If the docstring text references a citation more than once, sphinx includes links to each reference location instead of using the [1] as the link.

The in-line callouts are also placed as superscripts. I wonder if using the number scheme is confusing Citations with Footnotes, and if switching to an author-name scheme would improve the formatting.  Not a request for more work :)

wholmgren commented:
> This part of api.rst might have been overlooked when tmy.* was moved to iotools: https://pvlib-python.readthedocs.io/en/latest/api.html#tmy

Thanks. Let's remove it here.

> pvsystem.adrinverter doesn't have an entry in api.rst (other than through modelchain)

Good catch. Let's add it here.

> The in-line callouts are also placed as superscripts. I wonder if using the number scheme is confusing Citations with Footnotes, and if switching to an author-name scheme would improve the formatting. Not a request for more work :)

Does numpydoc/sphinx support this? I think this PR is a significant improvement on the current state, even if some things are not ideal.

wholmgren commented on docs/sphinx/source/conf.py:
```diff
@@ -320,3 +320,6 @@ def setup(app):
 ipython_warning_is_error = False
+
+# suppress "WARNING: Footnote [1] is not referenced." messages
+suppress_warnings = ['ref.footnote']
```
```suggestion
# https://github.com/pvlib/pvlib-python/issues/837
suppress_warnings = ['ref.footnote']
```

cwhanse commented:
@kevinsa5 @wholmgren  what other changes are planned / anticipated for this PR?

kevinsa5 commented:
Nothing planned from me, though I'm happy to make changes if there are any other requests.

wholmgren commented:
There's a line too long lint error in `pvsystem` that should be fixed. The `tmy` lint errors can be ignored.

Otherwise good to go.

kevinsa5 commented:
@wholmgren Thanks for pointing that out, I had missed it.  It is fixed along with one straggler docstring reference.

wholmgren commented:
thanks @kevinsa5 -- I really appreciate the maintenance help!
